kratos/processes/fast_transfer_between_model_parts_process.cpp

```diff
@@ -249,72 +243,76 @@ void FastTransferBetweenModelPartsProcess::ReplicateWithoutFlags()
 
     #pragma omp parallel
     {
-        const int thread_id = OpenMPUtils::ThisThread();
+        // Creating a buffer for parallel vector fill
+        NodesArrayType nodes_buffer_vector;
+        ElementsArrayType elements_buffer_vector;
+        ConditionsArrayType conditions_buffer_vector;
+        MasterSlaveConstraintArrayType constraints_buffer_vector;
 
         if (num_nodes != 0 && (mEntity == EntityTransfered::ALL || mEntity == EntityTransfered::NODES || mEntity == EntityTransfered::NODESANDELEMENTS || mEntity == EntityTransfered::NODESANDCONDITIONS)) {
+            const auto it_node_begin = mrOriginModelPart.NodesBegin();
```

Comment:
if you like you could add the nowait clause to the first two

Reply:
Ok